code/modules/power/singularity/containment_field.dm

```diff
@@ -98,16 +58,18 @@
 		bump_field(user)
 	return
 
-/obj/machinery/field/proc/clear_shock()
-	hasShocked = 0
-
-/obj/machinery/field/proc/bump_field(atom/movable/AM as mob|obj)
-	if(hasShocked)
-		return 0
-	hasShocked = 1

```

Comment:
if you remove hasShocked you're going to get an infinite loop of bump_field() when a mob is stuck between a containment field and a wall, getting thrown repeatedly. We had that var specifically to stop this. Look at the history of the file.